contracts/marketplace/src/execute.rs

```diff
@@ -1032,6 +1002,55 @@ fn only_operator(store: &dyn Storage, info: &MessageInfo) -> Result<Addr, Contra
     Ok(info.sender.clone())
 }
 
+enum HookReply {
+    Ask = 1,
+    Sale,
+    Bid,
+    CollectionBid,
+}
+
+impl From<u64> for HookReply {
+    fn from(item: u64) -> Self {
+        match item {
+            1 => HookReply::Ask,
+            2 => HookReply::Sale,
+            3 => HookReply::Bid,
+            4 => HookReply::CollectionBid,
+            _ => panic!("invalid reply type"),
+        }
+    }
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn reply(_deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractError> {
+    match HookReply::from(msg.id) {
+        HookReply::Ask => {
+            let res = Response::new()
+                .add_attribute("action", "ask-hook-failed")
+                .add_attribute("error", msg.result.unwrap_err());
+            Ok(res)
+        }
+        HookReply::Sale => {
+            let res = Response::new()
+                .add_attribute("action", "sale-hook-failed")
+                .add_attribute("error", msg.result.unwrap_err());
+            Ok(res)
+        }
+        HookReply::Bid => {
+            let res = Response::new()
+                .add_attribute("action", "bid-hook-failed")
+                .add_attribute("error", msg.result.unwrap_err());
+            Ok(res)
+        }
+        HookReply::CollectionBid => {
+            let res = Response::new()
+                .add_attribute("action", "collection-bid-hook-failed")
+                .add_attribute("error", msg.result.unwrap_err());
+            Ok(res)
+        }
+    }
+}
```

Comment:
This might give a better intent:

```
  let action = match HookReply::from(msg.id) {
    HookReply::Ask => { "ask-hook-failed" }
    HookReply::Sale => { "sale-hook-failed" }
    HookReply::Bid => { "bid-hook-failed" }
    HookReply::CollectionBid => { "collection-bid-hook-failed" }
  }
  let res = Response::new()
                 .add_attribute("action", action)
                 .add_attribute("error", msg.result.unwrap_err());
  Ok(res)
```